### DISCUSSION TITLE
Are there FFI bindings for D2? (allowing to call D2 from another language)

### DISCUSSION THREAD

**Discussion by dlight** [terrastruct/d2#1535] [Q&A]
Hi, I see that [D2 can be used as a library](https://github.com/terrastruct/d2#d2-as-a-library), but all examples are in Go. I'm not very familiar with the Go ecosystem but from what I gather elsewhere [I could create C bindings](https://github.com/vladimirvivien/go-cshared-examples) to call D2 from another language.

Is such FFI binding already available somewhere? I'm looking for some examples.

**Comment by alixander** [comment]
The closest perhaps is the effort to make D2 a WASM package. I started this but haven't finished (https://github.com/terrastruct/d2/compare/master...alixander:d2:wasm-build).

Is that what you're looking for? Or am I misunderstanding. @dlight

**Reply by zetashift** [reply]
I believe OP maybe wants to programmatically make diagrams using something else than Go, and instead of making a new parser from scratch they could bind to the Go API.

That being said, I'm very curious about the WASM package because I would like to show d2 diagrams in the browser from markdown docs.
